### PR TITLE
[KED-1555] Expose VERSION_FORMAT constant in Kedro IO library

### PR DESCRIPTION
**Pull request by deepyaman**
## Description
I’m currently trying to parse versions, and wanted to see if I could avoid hardcoding the version format myself.

This is my code:
```python3
from datetime import datetime
last_load_version = datetime.strptime(
    catalog.datasets.data_set_name.get_last_load_version(),  # Replace with `resolve_load_version` in future version of Kedro
    "%Y-%m-%dT%H.%M.%S.%fZ",
)
last_load_version
```

## Development notes
Manual test:
```python3
>>> from datetime import datetime
>>> 
>>> from kedro.io.core import VERSION_FORMAT
>>> 
>>> datetime.strptime(generate_timestamp(), VERSION_FORMAT)
datetime.datetime(2020, 3, 21, 21, 33, 59, 405000)
>>> 
```

~~I'll add a unit test verifying converted-back timestamp is correct (no microseconds) later; I'm currently having issues with local tests and CircleCI... =/ (all local test failures seemed completely unrelated)~~ Test added.

## Checklist

- [x] Read the [contributing](../CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](../RELEASE.md) file
- [x] Added tests to cover my changes

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Comment by deepyaman** [comment]
TL;DR: Tests still pass.
```bash
(kedro) BOS-178551-C02X31K9JHD4:kedro deepyaman$ pytest tests/io/test_data_catalog.py 
=========================================================================================== test session starts ============================================================================================
platform darwin -- Python 3.7.3, pytest-3.10.1, py-1.8.0, pluggy-0.12.0
rootdir: /Users/deepyaman/kedro, inifile: setup.cfg
plugins: requests-mock-1.7.0, cov-2.7.1, mock-1.10.4
collected 46 items                                                                                                                                                                                         

tests/io/test_data_catalog.py ..............................................                                                                                                                         [100%]

---------- coverage: platform darwin, python 3.7.3-final-0 -----------
Name                                                               Stmts   Miss  Cover   Missing
------------------------------------------------------------------------------------------------
kedro/__init__.py                                                      4      0   100%
kedro/__main__.py                                                      0      0   100%
kedro/cli/__init__.py                                                  2      2     0%   29-34
kedro/cli/cli.py                                                     215    215     0%   29-654
kedro/cli/jupyter.py                                                  14     14     0%   29-67
kedro/cli/utils.py                                                    64     64     0%   29-191
kedro/config/__init__.py                                               4      0   100%
kedro/config/config.py                                                78     62    21%   117-126, 149-192, 209-236, 251-261, 266-275
kedro/config/default_logger.py                                         8      0   100%
kedro/config/templated_config.py                                      45     33    27%   150-154, 178-183, 229-272
kedro/context/__init__.py                                              2      2     0%   29-33
kedro/context/context.py                                             206    206     0%   28-693
kedro/extras/__init__.py                                               0      0   100%
kedro/extras/datasets/__init__.py                                      1      0   100%
kedro/extras/datasets/biosequence/__init__.py                          2      2     0%   29-31
kedro/extras/datasets/biosequence/biosequence_dataset.py              42     42     0%   29-150
kedro/extras/datasets/dask/__init__.py                                 2      2     0%   29-31
kedro/extras/datasets/dask/parquet_dataset.py                         35     35     0%   29-146
kedro/extras/datasets/geopandas/__init__.py                            2      2     0%   28-31
kedro/extras/datasets/geopandas/geojson_dataset.py                    45     45     0%   29-166
kedro/extras/datasets/matplotlib/__init__.py                           2      2     0%   29-31
kedro/extras/datasets/matplotlib/matplotlib_writer.py                 48     48     0%   30-183
kedro/extras/datasets/networkx/__init__.py                             2      2     0%   29-32
kedro/extras/datasets/networkx/networkx_dataset.py                    47     47     0%   30-166
kedro/extras/datasets/pandas/__init__.py                              11      0   100%
kedro/extras/datasets/pandas/csv_blob_dataset.py                      48     29    40%   112-146, 149, 161-168, 171-173, 181-182, 188, 194-195
kedro/extras/datasets/pandas/csv_dataset.py                           47      5    89%   130, 162-163, 168-169
kedro/extras/datasets/pandas/excel_dataset.py                         54     36    33%   118-145, 148, 159-162, 165-175, 178-183, 186-187, 191-192
kedro/extras/datasets/pandas/feather_dataset.py                       43     26    40%   110-130, 133, 142-145, 148-156, 159-161, 164-165, 169-170
kedro/extras/datasets/pandas/gbq_dataset.py                           47     30    36%   111-129, 136, 145-146, 154, 162-167, 170-174
kedro/extras/datasets/pandas/hdf_dataset.py                           58     38    34%   118-142, 145, 156-172, 175-192, 195-200, 203-204, 208-209
kedro/extras/datasets/pandas/json_blob_dataset.py                     33     19    42%   110-125, 128, 139-145, 148
kedro/extras/datasets/pandas/json_dataset.py                          47     30    36%   110-132, 135, 145-148, 151-156, 159-164, 167-168, 172-173
kedro/extras/datasets/pandas/parquet_dataset.py                       49     19    61%   131, 134, 137, 147-150, 153-160, 163-168, 171-172, 176-177
kedro/extras/datasets/pandas/sql_dataset.py                           97     70    28%   72-87, 91-100, 104, 194-216, 219-225, 233-238, 241-246, 249-253, 320-341, 344-347, 350-355, 358
kedro/extras/datasets/pickle/__init__.py                               2      2     0%   29-31
kedro/extras/datasets/pickle/pickle_dataset.py                        51     51     0%   29-192
kedro/extras/datasets/spark/__init__.py                                4      4     0%   29-33
kedro/extras/datasets/spark/spark_dataset.py                          90     90     0%   29-276
kedro/extras/datasets/spark/spark_hive_dataset.py                     86     86     0%   29-321
kedro/extras/datasets/spark/spark_jdbc_dataset.py                     51     51     0%   29-189
kedro/extras/datasets/text/__init__.py                                 2      2     0%   29-31
kedro/extras/datasets/text/text_dataset.py                            46     46     0%   29-169
kedro/extras/datasets/yaml/__init__.py                                 2      2     0%   29-31
kedro/extras/datasets/yaml/yaml_dataset.py                            46     46     0%   29-166
kedro/extras/decorators/__init__.py                                    1      1     0%   29
kedro/extras/decorators/memory_profiler.py                            17     17     0%   29-86
kedro/extras/decorators/retry_node.py                                 26     26     0%   29-87
kedro/extras/logging/__init__.py                                       2      2     0%   29-33
kedro/extras/logging/color_logger.py                                  18     18     0%   28-122
kedro/extras/transformers/__init__.py                                  2      2     0%   29-31
kedro/extras/transformers/time_profiler.py                            17     17     0%   29-58
kedro/io/__init__.py                                                  15      0   100%
kedro/io/cached_dataset.py                                            39     25    36%   81-90, 93-94, 98-108, 111, 117-122, 125-126, 129, 134-136
kedro/io/core.py                                                     228     40    82%   196, 217, 222-228, 259, 266, 273, 292-296, 312-319, 322, 441-442, 524, 554, 574, 592-598, 601-603, 625-630, 647, 653-655
kedro/io/data_catalog.py                                             148     15    90%   182, 195, 327, 454, 580-596
kedro/io/data_catalog_with_default.py                                 28     15    46%   84-93, 111-116, 131-136, 167, 200
kedro/io/lambda_data_set.py                                           36     13    64%   63-64, 78-83, 86-91, 96, 99-102, 132
kedro/io/memory_data_set.py                                           47     10    79%   81, 92, 95, 125, 127, 150-155
kedro/io/partitioned_data_set.py                                     161     77    52%   154, 163-170, 174, 192, 197, 204-208, 211-214, 217-221, 224-237, 240-247, 250-255, 267, 270, 278-284, 397, 405, 409, 424, 427, 436-452, 460-461, 464-469, 472-483, 488-490
kedro/io/transformers.py                                               8      2    75%   53, 66
kedro/pipeline/__init__.py                                             3      3     0%   29-34
kedro/pipeline/decorators.py                                          19     19     0%   28-87
kedro/pipeline/node.py                                               196    196     0%   28-671
kedro/pipeline/pipeline.py                                           264    264     0%   28-959
kedro/runner/__init__.py                                               4      4     0%   29-35
kedro/runner/parallel_runner.py                                      108    108     0%   28-279
kedro/runner/runner.py                                                50     50     0%   28-189
kedro/runner/sequential_runner.py                                     28     28     0%   28-92
kedro/utils.py                                                        11      0   100%
kedro/versioning/__init__.py                                           2      0   100%
kedro/versioning/journal.py                                           41     12    71%   69-70, 108, 125-127, 136-138, 148-154
tests/__init__.py                                                      1      0   100%
tests/cli/__init__.py                                                  0      0   100%
tests/cli/conftest.py                                                 16     16     0%   29-61
tests/cli/test_cli.py                                                272    272     0%   28-494
tests/cli/test_cli_new.py                                            154    154     0%   29-401
tests/cli/test_jupyter.py                                             22     22     0%   29-73
tests/config/__init__.py                                               0      0   100%
tests/config/test_config.py                                          153    153     0%   29-332
tests/config/test_templated_config.py                                127    127     0%   29-411
tests/conftest.py                                                     39     14    64%   48-49, 70-84, 102
tests/context/__init__.py                                              0      0   100%
tests/context/test_context.py                                        374    374     0%   29-820
tests/extras/__init__.py                                               0      0   100%
tests/extras/datasets/__init__.py                                      0      0   100%
tests/extras/datasets/bioinformatics/__init__.py                       0      0   100%
tests/extras/datasets/bioinformatics/test_biosequence_dataset.py      53     53     0%   29-122
tests/extras/datasets/conftest.py                                     11     11     0%   29-58
tests/extras/datasets/dask/__init__.py                                 0      0   100%
tests/extras/datasets/dask/test_parquet_dataset.py                    87     87     0%   32-198
tests/extras/datasets/geojson/__init__.py                              0      0   100%
tests/extras/datasets/geojson/test_geojson_dataset.py                101    101     0%   28-229
tests/extras/datasets/matplotlib/__init__.py                           0      0   100%
tests/extras/datasets/matplotlib/test_matplotlib_writer.py           106    106     0%   30-238
tests/extras/datasets/networkx/__init__.py                             0      0   100%
tests/extras/datasets/networkx/test_networkx_dataset.py               91     91     0%   29-230
tests/extras/datasets/pandas/__init__.py                               0      0   100%
tests/extras/datasets/pandas/test_csv_blob_dataset.py                135    135     0%   31-369
tests/extras/datasets/pandas/test_csv_dataset.py                     110    110     0%   29-243
tests/extras/datasets/pandas/test_excel_dataset.py                    98     98     0%   29-216
tests/extras/datasets/pandas/test_feather_dataset.py                  89     89     0%   29-199
tests/extras/datasets/pandas/test_gbq_dataset.py                      73     73     0%   29-191
tests/extras/datasets/pandas/test_hdf_dataset.py                     110    110     0%   29-237
tests/extras/datasets/pandas/test_json_blob_dataset.py                69     69     0%   31-175
tests/extras/datasets/pandas/test_json_dataset.py                     92     92     0%   29-207
tests/extras/datasets/pandas/test_parquet_dataset.py                  99     99     0%   29-224
tests/extras/datasets/pandas/test_sql_dataset.py                     156    156     0%   31-319
tests/extras/datasets/pickle/__init__.py                               0      0   100%
tests/extras/datasets/pickle/test_pickle_dataset.py                  103    103     0%   29-225
tests/extras/datasets/text/__init__.py                                 0      0   100%
tests/extras/datasets/text/test_text_dataset.py                       93     93     0%   29-207
tests/extras/datasets/yaml/__init__.py                                 0      0   100%
tests/extras/datasets/yaml/test_yaml_dataset.py                       98     98     0%   29-211
tests/extras/decorators/__init__.py                                    0      0   100%
tests/extras/decorators/test_memory_profiler.py                       33     33     0%   29-83
tests/extras/decorators/test_retry_node.py                            14     14     0%   28-47
tests/extras/logging/__init__.py                                       0      0   100%
tests/extras/logging/test_color_logger.py                             10     10     0%   29-44
tests/extras/transformers/__init__.py                                  0      0   100%
tests/extras/transformers/test_time_profiler.py                       27     27     0%   29-71
tests/io/__init__.py                                                   0      0   100%
tests/io/test_cached_dataset.py                                       80     80     0%   29-172
tests/io/test_data_catalog.py                                        270      0   100%
tests/io/test_data_catalog_with_default.py                            50     50     0%   29-149
tests/io/test_incremental_dataset.py                                 170    170     0%   28-532
tests/io/test_lambda_data_set.py                                     105    105     0%   29-222
tests/io/test_memory_data_set.py                                     123    123     0%   29-254
tests/io/test_partitioned_dataset.py                                 227    227     0%   28-501
tests/io/test_transformers.py                                         92     92     0%   29-161
tests/pipeline/__init__.py                                             0      0   100%
tests/pipeline/test_decorators.py                                     42     42     0%   29-92
tests/pipeline/test_node.py                                          218    218     0%   28-457
tests/pipeline/test_node_run.py                                       87     87     0%   31-169
tests/pipeline/test_pipeline.py                                      351    351     0%   28-831
tests/pipeline/test_pipeline_from_missing.py                         103    103     0%   29-249
tests/pipeline/test_pipeline_integration.py                           28     28     0%   29-106
tests/pipeline/test_pipeline_transform.py                             72     72     0%   28-214
tests/pipeline/test_pipeline_with_transcoding.py                     150    150     0%   28-420
tests/runner/__init__.py                                               0      0   100%
tests/runner/test_parallel_runner.py                                 144    144     0%   30-356
tests/runner/test_sequential_runner.py                               127    127     0%   30-320
tests/test_cli_logging_setup.py                                       14     14     0%   29-52
tests/test_utils.py                                                   18     18     0%   29-58
------------------------------------------------------------------------------------------------
TOTAL                                                               8790   7662    13%
Coverage XML written to file coverage.xml


======================================================================================== 46 passed in 8.03 seconds =========================================================================================
```

**Comment by deepyaman** [comment]
Replaced by #323
